<?xml version="1.0"?>
<launch>
  <arg name="gui" default="true"/>
  <arg name="paused" default="false"/>
  <arg name="verbose" default="false"/>
  <arg name="debug" default="false"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find ekg_auv_testing)/worlds/water_sim.world"/>
    <arg name="extra_gazebo_args" value=""/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="verbose" value="$(arg verbose)"/>
  </include>

  <!-- Beacons -->
  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_1"/>
    <arg name="bid" value="1"/>
    <arg name="transceiver_id" value="141"/>
    <!--arg name="transponder_id" value="$(arg bid)"/-->
    <arg name="x" value="0"/>
    <arg name="y" value="0"/>
    <arg name="z" value="-0.2"/>
  </include>

  <!--include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_2"/>
    <arg name="bid" value="2"/>
    <arg name="transceiver_id" value="142"/>
    <arg name="x" value="0"/>
    <arg name="y" value="20"/>
    <arg name="z" value="-0.2"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_3"/>
    <arg name="bid" value="3"/>
    <arg name="x" value="20"/>
    <arg name="y" value="0"/>
    <arg name="z" value="-0.2"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_4"/>
    <arg name="bid" value="4"/>
    <arg name="x" value="20"/>
    <arg name="y" value="20"/>
    <arg name="z" value="-0.2"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_10"/>
    <arg name="bid" value="10"/>
    <arg name="x" value="0"/>
    <arg name="y" value="0"/>
    <arg name="z" value="-40"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_11"/>
    <arg name="bid" value="11"/>
    <arg name="x" value="0"/>
    <arg name="y" value="20"/>
    <arg name="z" value="-40"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_12"/>
    <arg name="bid" value="12"/>
    <arg name="x" value="20"/>
    <arg name="y" value="0"/>
    <arg name="z" value="-40"/>
  </include>

  <include file="$(find ekg_auv_testing)/launch/upload_beacon_config.launch">
    <arg name="robot_name" value="beacon_13"/>
    <arg name="bid" value="13"/>
    <arg name="x" value="20"/>
    <arg name="y" value="20"/>
    <arg name="z" value="-40"/>
  </include-->
</launch>
